prospero-cli/src/main/java/org/wildfly/prospero/cli/commands/UpdateCommand.java

```diff
@@ -471,4 +479,49 @@ public static Path detectProsperoInstallationPath() throws ArgumentParsingExcept
         return Paths.get(modulePath).toAbsolutePath().getParent();
     }
 
+
+    public static void compareManifests(List<ManifestVersionRecord.MavenManifest> mavenManifests, List<Artifact> manifestUpdates ) {
+
+        List<ManifestVersionRecord.MavenManifest> intersectingManifests = new ArrayList<>();
+
+        // Find intersecting manifests (those that are present in both mavenManifests and manifestUpdates)
+        for (ManifestVersionRecord.MavenManifest installedManifest : mavenManifests) {
+            for (Artifact update : manifestUpdates) {
+                if (update.getGroupId().equals(installedManifest.getGroupId()) &&
+                        update.getArtifactId().equals(installedManifest.getArtifactId())) {
+                    intersectingManifests.add(installedManifest); // Add to intersection list
+                    break;
+                }
+            }
+        }
+
+        // For each intersecting manifest, compare the version
+        for (ManifestVersionRecord.MavenManifest installedManifest : intersectingManifests) {
```

Comment:
you don't need the `intersectingManifest` list. Just iterate over mavenManifests. You're already ignoring ones that are not in `updateArtifacts`.